server/src/rpc/mod.rs

```diff
@@ -13,6 +14,9 @@ pub mod wal_log;
 
 pub async fn run_rpc_server<F>(
     addr: SocketAddr,
+    tls: bool,
+    cert: Option<PathBuf>,
+    key: Option<PathBuf>,
```

Comment:
Perhaps these should be packed into a single `Option<(PathBuf, PathBuf)>` parameter, since key and cert should be assigned only if tls is true?